website/docs/r/encryption_at_rest.html.markdown

```diff
@@ -59,13 +62,55 @@ resource "mongodbatlas_encryption_at_rest" "test" {
 
 **NOTE**  if using the two resources path for cloud provider access, `cloud_provider_access_setup` and `cloud_provider_access_authorization`, you may need to define a `depends_on` statement for these two resources, because terraform is not able to infer the dependency.
 
-```hcl
+```terraform
 resource "mongodbatlas_encryption_at_rest" "default" {
   (...)
   depends_on = [mongodbatlas_cloud_provider_access_setup.<resource_name>, mongodbatlas_cloud_provider_access_authorization.<resource_name>]
 }
 ```
 
+## Example: Configuring encryption at rest using customer key management in Azure and then creating a cluster
+
+The configuration of encryption at rest with customer key management, `mongodbatlas_encryption_at_rest`, needs to be completed before a cluster is created in the project. Force this wait by using an implicit dependency via `project_id` as shown in the example below.
+
+```terraform
+resource "mongodbatlas_encryption_at_rest" "example" {
```

Comment:
You tested this right?

Reply:
Yes for the example I tested manually with and without using implicit dependency, without dependency got some issues related to waiting and with dependency I didn't had any issues for creating/destroying.

Reply:
@coderGo93 did you test the option I noted, setting the clusters to None?  We should probably make sure there's no timing issue with that and Terraform as well.

Reply:
@themantissa Not like that, will do that and let you know if I had issues or not.

Reply:
Hi @themantissa , I tested manually again with setting None in cluster, it worked without any issues.
Steps: 
1. Create a encryption then cluster with dependency
2. Set `encryption_at_rest_provider` to None in cluster then terraform apply to update
3. Then destroy the encryption at rest resource without destroying the cluster(setting count = 0 in encryption at rest resource)

Reply:
Well done @coderGo93, just what we needed.  Thank you